docs/docs/flink-table-maintenance.md

```diff
@@ -0,0 +1,268 @@
+    .add(RewriteDataFiles.builder()
+        .scheduleOnDataFileCount(10)
+        .targetFileSizeBytes(128 * 1024 * 1024)
+        .partialProgressEnabled(true)
+        .partialProgressMaxCommits(10))
+    .append();
+
+env.execute("Table Maintenance Job");
+```
+
+## Configuration Options
+
+### TableMaintenance Builder
+
+| Method | Description | Default |
+|--------|-------------|---------|
+| `uidSuffix(String)` | Unique identifier suffix for the job | Random UUID |
+| `rateLimit(Duration)` | Minimum interval between task executions | 60 seconds |
+| `lockCheckDelay(Duration)` | Delay for checking lock availability | 30 seconds |
+| `parallelism(int)` | Default parallelism for maintenance tasks | System default |
+| `maxReadBack(int)` | Max snapshots to check during initialization | 100 |
+
+### ExpireSnapshots Configuration
+
+| Method | Description | Default Value | Type |
+|--------|-------------|---------------|------|
+| `maxSnapshotAge(Duration)` | Maximum age of snapshots to retain | No limit | Duration |
+| `retainLast(int)` | Minimum number of snapshots to retain | 1 | int |
+| `deleteBatchSize(int)` | Number of files to delete in each batch | 1000 | int |
+| `scheduleOnCommitCount(int)` | Trigger after N commits | No automatic scheduling | int |
+| `scheduleOnDataFileCount(int)` | Trigger after N data files | No automatic scheduling | int |
+| `scheduleOnDataFileSize(long)` | Trigger after total data file size (bytes) | No automatic scheduling | long |
+| `scheduleOnIntervalSecond(long)` | Trigger after time interval (seconds) | No automatic scheduling | long |
+| `parallelism(int)` | Parallelism for this specific task | Inherits from TableMaintenance | int |
+
+
+
+### RewriteDataFiles Configuration
+
+| Method | Description | Default Value | Type |
+|--------|-------------|---------------|------|
+| `targetFileSizeBytes(long)` | Target size for rewritten files | Table property or 512MB | long |
+| `partialProgressEnabled(boolean)` | Enable partial progress commits | false | boolean |
+| `partialProgressMaxCommits(int)` | Maximum commits for partial progress | 10 | int |
+| `scheduleOnCommitCount(int)` | Trigger after N commits | 10 | int |
+| `scheduleOnDataFileCount(int)` | Trigger after N data files | 1000 | int |
+| `scheduleOnDataFileSize(long)` | Trigger after total data file size (bytes) | 100GB | long |
+| `scheduleOnIntervalSecond(long)` | Trigger after time interval (seconds) | 600 (10 minutes) | long |
+| `maxRewriteBytes(long)` | Maximum bytes to rewrite per execution | Long.MAX_VALUE | long |
+| `parallelism(int)` | Parallelism for this specific task | Inherits from TableMaintenance | int |
+
+## Flink Configuration Options
+
+You can also configure maintenance behavior through Flink configuration:
+
+| Configuration Key | Description | Default Value | Type |
+|-------------------|-------------|---------------|------|
+| `iceberg.maintenance.rate-limit-seconds` | Rate limit in seconds | 60 | long |

```

Comment:
Could this be just another column like in https://iceberg.apache.org/docs/nightly/flink-configuration/#read-options?